src/rtctools_heat_network/esdl/esdl_model_base.py

```diff
@@ -1,32 +1,51 @@
 import logging
+from typing import Dict
 
 import esdl
 from esdl import InPort
 
-
+from rtctools_heat_network.esdl.asset_to_component_base import _AssetToComponentBase
 from rtctools_heat_network.pycml import Model as _Model
 
+from ._exceptions import _RetryLaterException, _SkipAssetException
+
 logger = logging.getLogger("rtctools_heat_network")
 
 
 RETRY_LOOP_LIMIT = 100
 
 
-class _RetryLaterException(Exception):
-    pass
-
-
-class _SkipAssetException(Exception):
-    pass
-
-
 class _ESDLModelBase(_Model):
+    """
+    This is the ESDL base class that is used to convert from the esdl parsed assets to the pycml
+    model description. In the base class we specify how the network is connected, meaning which
+    asset is connected to which asset as this is the same for different model types (both
+    Heat and QTH). Per model type we have a specialization for the converting of the assets as
+    different models might need different information from the esdl file.
+    """
+
     primary_port_name_convention = "prim"
     secondary_port_name_convention = "sec"
 
-    def _esdl_convert(self, converter, assets, prefix):
+    def _esdl_convert(self, converter: _AssetToComponentBase, assets: Dict, prefix: str) -> None:
+        """
+        In this function we convert the esdl parsed assets and instantiate the pycml objects for
+        those assets. We use the converter to create those pycml objects. Afterwards we look at the
+        connections specified in the esdl and create the relevant maps between ports to then also
```

Comment:
" Afterwards we look at the connections specified ... " does this also happen in this function or somewhere else, because it sounds like it might be happen in this function after objects have been created?

Reply:
no happens here

Reply:
fix